tests/phpunit/tests/image/functions.php

```diff
@@ -726,6 +726,68 @@ public function test_wp_generate_attachment_metadata_pdf() {
 		remove_filter( 'wp_upload_image_mime_transforms', '__return_empty_array' );
 	}
 
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_generate_attachment_metadata_pdf_webp_mime() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'application/pdf' ) ) ) {
+			$this->markTestSkipped( 'Rendering PDFs is not supported on this system.' );
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
+
+		// Use legacy JPEG output and WebP.
+		add_filter(
+			'wp_upload_image_mime_transforms',
+			function () {
+				return array(
+					'image/jpeg'      => array( 'image/jpeg', 'image/webp' ),
+					'image/webp'      => array( 'image/webp', 'image/jpeg' ),
+					'application/pdf' => array( 'image/webp', 'image/jpeg' ),
+				);
+			}
+		);
+
+		$orig_file = DIR_TESTDATA . '/images/wordpress-gsoc-flyer.pdf';
+		$test_file = get_temp_dir() . 'wordpress-gsoc-flyer.pdf';
+		copy( $orig_file, $test_file );
+
+		$editor = wp_get_image_editor( $test_file );
+		if ( is_wp_error( $editor ) ) {
+			$this->markTestSkipped( $editor->get_error_message() );
+		}
+
+		$attachment_id = $this->factory->attachment->create_object(
+			$test_file,
+			0,
+			array(
+				'post_mime_type' => 'application/pdf',
+			)
+		);
+
+		$this->assertNotEmpty( $attachment_id );
+
+		$temp_dir = get_temp_dir();
+
+		$metadata = wp_generate_attachment_metadata( $attachment_id, $test_file );
+
+		$this->assertNotEmpty( $metadata );
+		$this->assertArrayHasKey( 'sizes', $metadata );
+
+		foreach ( $metadata['sizes'] as $size ) {
```

Comment:
we should validate that webp versions were actually output (or all the types specified in `wp_upload_image_mime_transforms`) when supported

Reply:
if we want to test the output then some of the environments doesn't actually support WebP but it prioritise the jpeg over WebP so the metadata generated has WebP in sources but it doesn't actually output it unless filtered via hooks. for example in 7.0 WebP is supported but it doesn't prioritise it and main versions generated are jpeg ones.

Reply:
I have put checks for mime types to verify them in `sources` attributes.